mirascope/core/cohere/_utils/setup_call.py

```diff
@@ -50,9 +50,14 @@ def setup_call(
     if len(messages) > 1:
         call_kwargs["chat_history"] = messages[:-1]
     if json_mode:
-        messages[-1].message += _utils.json_mode_content(
-            tool_types[0] if tool_types else None
+        # Cannot mutate ChatMessage in place
+        last_message = ChatMessage(
+            role=messages[-1].role,  # type: ignore
+            message=messages[-1].message
+            + _utils.json_mode_content(tool_types[0] if tool_types else None),
+            tool_calls=messages[-1].tool_calls,
         )
+        messages[-1] = last_message
```

Comment:
why not just assign the value directly rather than assign to `last_message` and then `messages[-1]`?

Reply:
done